### ISSUE TITLE
PHP Hammer: Class reference format must be "ExampleTrait"

### ISSUE THREAD

**Issue by asgraf** [hammer-tools/php-hammer#20]
I have `Class self-reference format normalization` set to `Prefer ClassName reference` in my project.

```php
trait ExampleTrait
{
    public function exampleFunction(): self 
    {
        // some code
        return $this;
    }
}
```
The example above is currently emitting the following warning:
`PHP Hammer: class reference format must be "ExampleTrait"`
However, trait usage is not allowed there, and replacing `self` with `ExampleTrait` using the quick-fix option will result in a fatal error.
This is because the function will return an instance of the class that the trait was applied to, rather than the trait itself.

**Comment by rentalhost** [comment]
You are right! I will mark it as a bug, should be fixed this month. Thanks!

**Comment by rentalhost** [comment]
Thank you for taking the time to write about this issue. The fix has been implemented for version 0.23.0 and is now available on the [**Releases**](https://github.com/hammer-tools/php-hammer/releases) page. It will soon be accessible on the JetBrains Marketplace as well, allowing you to update directly through PhpStorm.

Starting now, this inspection will ignore _traits_, as this feature can alter expected behavior, as demonstrated by your case.
